src/System.Buffers.Experimental/System/Buffers/BufferExtensions.cs

```diff
@@ -117,5 +118,107 @@ internal static int IndexOfStraddling(this ReadOnlySpan<byte> first, IReadOnlyMe
 
             return -1;
         }
+
+        static readonly int s_longSize = Vector<ulong>.Count;
+        static readonly int s_byteSize = Vector<byte>.Count;
+
+        public static int IndexOfVectorized(this Span<byte> buffer, byte value)
+        {
+            Debug.Assert(s_longSize == 4 || s_longSize == 2);
+
+            var byteSize = s_byteSize;
```

Comment:
Jit team told me off for not using `Vector<byte>.Count` directly; as they can't infer intrincs like auto loop unrolling https://github.com/dotnet/coreclr/pull/8001